streams/actions/src/main/java/com/hortonworks/streamline/streams/actions/topology/state/TopologyStates.java

```diff
@@ -0,0 +1,165 @@
+        @Override
+        public void deploy(TopologyContext context) throws Exception {
+            try {
+                context.setCurrentAction("Submitting topology to streaming engine");
+                Topology topology = context.getTopology();
+                TopologyDag dag = topology.getTopologyDag();
+                if (dag == null) {
+                    throw new IllegalStateException("Topology dag not set up");
+                }
+                TopologyActions topologyActions = context.getTopologyActions();
+                String mavenArtifacts = context.getMavenArtifacts();
+                topologyActions.deploy(CatalogToLayoutConverter.getTopologyLayout(topology, dag), mavenArtifacts, context);
+                context.setState(TOPOLOGY_STATE_DEPLOYED);
+                context.setCurrentAction("Topology deployed");
+            } catch (Exception ex) {
+                LOG.error("Error while trying to deploy the topology in the streaming engine", ex);
+                context.setState(TOPOLOGY_STATE_INITIAL);
+                context.setCurrentAction("Topology submission failed due to: " + ex);
+                throw ex;
+            }
+        }
+    };
+
+    public static final TopologyState TOPOLOGY_STATE_DEPLOYED = new TopologyState() {
+        @Override
+        public void kill(TopologyContext context) throws Exception {
+            try {
+                context.setCurrentAction("Killing topology");
+                Topology topology = context.getTopology();
+                TopologyActions topologyActions = context.getTopologyActions();
+                topologyActions.kill(CatalogToLayoutConverter.getTopologyLayout(topology));
+                context.setState(TOPOLOGY_STATE_INITIAL);
+                context.setCurrentAction("Topology killed");
+            } catch (Exception ex) {
+                LOG.error("Error while trying to kill the topology", ex);
+                context.setCurrentAction("Killing the topology failed due to: " + ex);
+                throw ex;
+            }
+        }
+
+        @Override
+        public void suspend(TopologyContext context) throws Exception {
+            try {
+                context.setCurrentAction("Suspending topology");
+                Topology topology = context.getTopology();
+                TopologyActions topologyActions = context.getTopologyActions();
+                topologyActions.suspend(CatalogToLayoutConverter.getTopologyLayout(topology));
+                context.setState(TOPOLOGY_STATE_SUSPENDED);
+                context.setCurrentAction("Topology suspended");
+            } catch (Exception ex) {
+                LOG.error("Error while trying to suspend the topology", ex);
+                context.setCurrentAction("Suspending the topology failed due to: " + ex);
+                throw ex;
+            }
+        }
+    };
+
+    public static final TopologyState TOPOLOGY_STATE_SUSPENDED = new TopologyState() {
```

Comment:
Would be better to allow `kill` while suspending without going through `resume`.